### DISCUSSION TITLE
[Unstable] Very early depth warnings

### DISCUSSION THREAD

**Discussion by ubertpendragon** [FakeFishGames/Barotrauma#14015] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [X] My issue happened while using mods.

### What happened?

For some reason the game throws very early depth warnings on modded map:

1061m warning for 3500 max depth resistance.

![image](https://github.com/FakeFishGames/Barotrauma/assets/73229309/85e1df2b-a7cc-40fc-a132-62403776ac31)

The new formula for calculating depth is way too aggressive

(https://github.com/FakeFishGames/Barotrauma/blob/e5694a08cfb7224d4a90194de78f8a6de19758cb/Barotrauma/BarotraumaClient/ClientSource/Map/Map/Map.cs#L1119-L1139)

Something to do with ?

`- Fixed a potential softlock if you accidentally start a round in a level that's too deep for the sub. The crush depth warnings on the campaign map now take the start/end position of the level into account, and the sub doesn't take depth damage in the first minute of the round, giving you a chance to return if you end up in a level that's too deep.`

### Reproduction steps

https://steamcommunity.com/sharedfiles/filedetails/?id=2532991202

Start new campaign

### Bug prevalence

Just once

### Single player or multiplayer?

Single player

### Version

v1.4.6.0 (Blood in the Water Update, hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by ubertpendragon** [comment]
Maybe related to

8daddc357b: Fixed potential softlock if you accidentally start a round in a level that's too deep for the sub (FakeFishGames/Barotrauma#13506):
- The crush depth warnings on the campaign map now take the start/end position of the level into account (showing the depth based on whichever is deeper).
- The sub doesn't take depth damage in the first minute of the round, giving you a chance to return if you end up in a level that's too deep.

3500m depth warning even though the lowest start/end position is at 2500m, so there's a whole 1000m of difference between low point and crush depth, the warning is not warranted

![image](https://github.com/FakeFishGames/Barotrauma/assets/73229309/70d7f128-fd5c-4d6e-9010-6b6615f4a13b)
![image](https://github.com/FakeFishGames/Barotrauma/assets/73229309/569925b2-b31a-491b-916f-a25f504900e8)
![image](https://github.com/FakeFishGames/Barotrauma/assets/73229309/6a4bfce2-50c7-406e-8789-0d5ac653f6a8)

**Comment by SomeRandomNoobKekeke** [comment]
Wait, is it in 1.4.6.0, 1.5.8.0 or some unstable :BaroDev:
i mean, it doesn't look like they touched this code in 1.6.0.0 again
[The fix](https://github.com/FakeFishGames/Barotrauma/commit/230d1b6e7887cd354bdf0bacc3952c2d3068c6b0#diff-0f2d51ebe94eed49df6fff1f16a0e6c09fcb23fcf76f4621f9ba1a14c02a6e2f) (might take a while to jump to that file)

I believe the logic is this:
If start or end is below crush depth then it's red
If crush depth is above abyss (< initial depth + level height) then it's yellow

Are you sure your level isn't 10km tall?  :BaroDev:

**Comment by ubertpendragon** [comment]
Ah well I adjusted the level values in the mod using https://www.desmos.com/calculator/eh6gqigsdq
